hips/draw/paint.py

```diff
@@ -109,30 +112,17 @@ def projection(self, tile: HipsTile) -> ProjectiveTransform:
         pt.estimate(src, dst)
         return pt
 
-    def _fetch_tiles(self) -> Iterator[HipsTile]:
-        """Generator function to fetch HiPS tiles from a remote URL."""
-        if self.progress_bar:
-            from tqdm import tqdm
-            tile_indices = tqdm(self.tile_indices, desc='Fetching tiles')
-        else:
-            tile_indices = self.tile_indices
-
-        for healpix_pixel_index in tile_indices:
-            tile_meta = HipsTileMeta(
-                order=self.draw_hips_order,
-                ipix=healpix_pixel_index,
-                frame=self.hips_survey.astropy_frame,
-                file_format=self.tile_format,
-            )
-            url = self.hips_survey.tile_url(tile_meta)
-            tile = HipsTile.fetch(tile_meta, url)
-            yield tile
-
     @property
     def tiles(self) -> List[HipsTile]:
         """List of `~hips.HipsTile` (cached on multiple access)."""
+        tile_fetcher = HipsTileFetcher(self.tile_indices, self.draw_hips_order, self.hips_survey,
+                                       self.tile_format, self.progress_bar)
         if self._tiles is None:
-            self._tiles = list(self._fetch_tiles())
+            if self.use_aiohttp:
+                import asyncio
+                self._tiles = asyncio.get_event_loop().run_until_complete(tile_fetcher.tiles_aiohttp)
```

Comment:
The `import asyncio` shouldn't appear in `paint.py`, users of the fetcher shouldn't have to know anything about the implementation.
Please move that to the fetcher, and expose from the fetcher only a synchronous API, where the user requests a list of tiles, the call blocks, and is handed back a list of tiles (or error info about missing requests / tiles) when the call comes back.